lib/spack/spack/cmd/view.py

```diff
@@ -241,7 +241,11 @@ def link_one(spec, path, link=os.symlink, verbose=False):
                     continue    # silence these
                 tty.warn("Skipping existing file: %s" % dst)
                 continue
-            link(src, dst)
+            if os.path.islink(src):
+                linkto = os.readlink(src)
```

Comment:
Should we resolve the path that is returned here? Or are we trusting that the path it points to will be linked locally as well?

Reply:
In practice I ended up with copies of all libraries, even ones that were symbolic links within the install directory. For example, libfoo.so ->libfoo.1.so->libfoo.1.2.3.so would end up with three identical copies of libfoo.1.2.3.so.

Reply:
View should never copy anyways. It should hardlink or symlink. What was the link count on the "identical copies"?

Reply:
Hmm, they shouldn't be appearing as real files if they're hardlinks to symlinks. Here's what I see:

```
total 0
lrwxrwxrwx. 2 boeckb boeckb 4 Oct 19 15:30 hard-to-symlink -> real
-rw-r-----. 1 boeckb boeckb 0 Oct 19 15:30 real
lrwxrwxrwx. 2 boeckb boeckb 4 Oct 19 15:30 symlink -> real
```

Note the `2` in the second column indicating that there are multiple names for the same file content.

Reply:
It's been so long I don't recall what prompted me to fix this.